processors/filterprocessor/expression/expression.go

```diff
@@ -0,0 +1,65 @@
+package expression
+
+import (
+	"fmt"
+	"regexp"
+)
+
+// Type is the type of the filter (i.e. const, regex, etc)
+type Type string
+
+const (
+	// ConstFilter a filter that looks at a constant string in its entirety
+	ConstFilter Type = "const"
+	// RegexFilter a filter that applies regex rules to the matching
+	RegexFilter Type = "regex"
+)
+
+// TypeToFunctionMap maps the expression-type with the needed function for the expression.
+// For instance the const or regex expression-type might need the String() function
+// Can't make this consts because there are no constant maps in go...
+var TypeToFunctionMap = map[Type]string{
+	ConstFilter: "String",
+	RegexFilter: "String",
+}
+
+// Interface the expression interface
+type Interface interface {
```

Comment:
The constructor is `MakeExpression`, should the name of this interface be Expression?